coupler/atmos_ocean_fluxes.F90

```diff
@@ -535,172 +546,345 @@ subroutine atmos_ocean_fluxes_init(gas_fluxes, gas_fields_atm, gas_fields_ice, v
+              & 'atm/name', index = m)
+          gas_fields_atm%bc_r4(n)%field(m)%override = .false.
+          gas_fields_atm%bc_r4(n)%field(m)%mean     = .false.
+        enddo
+
+        gas_fields_ice%bc_r4(n)%name = name
+        do m = 1, fm_util_get_length(trim(flux_list) // 'ice/name')
+          gas_fields_ice%bc_r4(n)%field(m)%name = trim(name) // "_" // fm_util_get_string(trim(flux_list) // &
+                                          &  'ice/name', index = m)
+          gas_fields_ice%bc_r4(n)%field(m)%override = .false.
+          gas_fields_ice%bc_r4(n)%field(m)%mean     = .false.
+        enddo
+
+        ! Save the units.
+        do m = 1, fm_util_get_length(trim(flux_list) // 'flux/name')
+          gas_fluxes%bc_r4(n)%field(m)%units =&
+              & fm_util_get_string(trim(fm_util_get_string(trim(flux_list) // 'flux/name', index = m)) // &
+              &  '-units', scalar = .true.)
+        enddo
+        do m = 1, fm_util_get_length(trim(flux_list) // 'atm/name')
+          gas_fields_atm%bc_r4(n)%field(m)%units =&
+              & fm_util_get_string(trim(fm_util_get_string(trim(flux_list) // 'atm/name', index = m)) // '-units')
+        enddo
+        do m = 1, fm_util_get_length(trim(flux_list) // 'ice/name')
+          gas_fields_ice%bc_r4(n)%field(m)%units =&
+              & fm_util_get_string(trim(fm_util_get_string(trim(flux_list) // 'ice/name', index = m)) // '-units')
+        enddo
+
+        ! Save the long names.
+        do m = 1, fm_util_get_length(trim(flux_list) // 'flux/name')
+          gas_fluxes%bc_r4(n)%field(m)%long_name =&
+              & fm_util_get_string(trim(fm_util_get_string(trim(flux_list) // 'flux/name', index = m)) // &
+              &  '-long_name', scalar = .true.)
+          gas_fluxes%bc_r4(n)%field(m)%long_name = trim(gas_fluxes%bc_r4(n)%field(m)%long_name) // ' for ' // name
+        enddo
+        do m = 1, fm_util_get_length(trim(flux_list) // 'atm/name')
+          gas_fields_atm%bc_r4(n)%field(m)%long_name =&
+              & fm_util_get_string(trim(fm_util_get_string(trim(flux_list) // 'atm/name', index = m)) // '-long_name')
+          gas_fields_atm%bc_r4(n)%field(m)%long_name = trim(gas_fields_atm%bc_r4(n)%field(m)%long_name) // ' for ' // name
+        enddo
+        do m = 1, fm_util_get_length(trim(flux_list) // 'ice/name')
+          gas_fields_ice%bc_r4(n)%field(m)%long_name =&
+              & fm_util_get_string(trim(fm_util_get_string(trim(flux_list) // 'ice/name', index = m)) // '-long_name')
+          gas_fields_ice%bc_r4(n)%field(m)%long_name = trim(gas_fields_ice%bc_r4(n)%field(m)%long_name) // ' for ' // name
+        enddo
+
+        ! Save the atm_tr_index.
+        gas_fluxes%bc_r4(n)%atm_tr_index = fm_util_get_integer('atm_tr_index', scalar = .true.)
+
+        ! Save the molecular weight.
+        gas_fluxes%bc_r4(n)%mol_wt = fm_util_get_real('mol_wt', scalar = .true.)
```

Comment:
should there be a real,r4_kind) around fm_util_get_real?  I think fm_util_get_real returns an r8_kind

Reply:
I just left the mol_wt and param fields as r8_kind regardless, this way they are always the value returned by the field manager.